src/test/fuzz/decode_tx.cpp

```diff
@@ -24,8 +24,7 @@ void test_one_input(const std::vector<uint8_t>& buffer)
     if (result_try_witness_and_maybe_no_witness) {
         assert(result_try_no_witness || result_try_witness);
     }
-    // if (result_try_no_witness) { // Uncomment when https://github.com/bitcoin/bitcoin/pull/17775 is merged
-    if (result_try_witness) { // Remove stop-gap when https://github.com/bitcoin/bitcoin/pull/17775 is merged
+    if (result_try_no_witness) {
         assert(result_try_witness_and_maybe_no_witness);
     }
```

Comment:
nit: to me it made more sense what this was doing when stepping back and rewriting it like this:
```c++
assert(!result_none);
if (result_try_witness) {
    assert(result_try_witness_and_maybe_no_witness);
} if (result_try_no_witness) {
    assert(result_try_witness_and_maybe_no_witness);
}
```
idk if it would be more logical to use `else if` here

Reply:
or maybe even cleaner:
```c++
assert(!result_none);
if (result_try_witness || result_try_no_witness) {
    assert(result_try_witness_and_maybe_no_witness);
}
```

Reply:
maybe I'm not thinking about this right though.. I do see how this can be different. Fuzzing is new to me, interested to see what you think (or if I'm wrong here)